### PR TITLE
Add code to overlay banding locations on range map

### PR DESCRIPTION
**Pull request by ezylstra**
I added code at the end of the script to overlay banding locations on the range map and zoom into the region where BTLH banding stations are located.  I also created an example of how you could label those banding locations on the map, but only did this for a few random stations because many are so close together.  

I also had a few suggestions for modifying other parts of the code - just where I ran into errors when I tried to execute the script on my computer.  All of these are very minor.  

I included my initials (EZ) anywhere I suggested changes or additions to the code.  Everything new is commented out, except for the mapping stuff which is at the end (and would be easy to edit/remove as you see fit).

### CODE REVIEW

File: scripts/BTLH_data_exploration.R [hunk 1]
```diff
@@ -146,6 +154,9 @@ BTLH_sites <- BTLH_HMN %>%
             N.Females = length(unique(Band.Number[Sex == "F"]))) %>% 
   arrange(N.Captures) %>% 
   as.data.frame
+# EZ: a few sites have the same location name except there's a 1 after it.
+# Can the data from these sites be combined?  (if, for example, the banding
+# station was moved just a little ways away but it's basically the same site)
```
Comment (GabySamaniego) [inline scripts/BTLH_data_exploration.R:160]: I had the same thought, and it is actually one of my questions for tomorrow. I think the sites with the 1 are very close, so merging them should be fine.

File: scripts/BTLH_data_exploration.R [hunk 2]
```diff
@@ -14,9 +14,17 @@ raw_data <- read.csv("data/updated_raw_data.csv")
 
 # Capitalize all characters and factors across data frame 
 band_data <- mutate_all(raw_data, .funs=toupper)
+# EZ: Getting an error when I run this related to TailMid column
+# If you're also getting error, suggest replacing above with the following:
+# band_data <- raw_data %>%
+#   mutate(across(!TailMid, .funs = toupper))
```
Comment (GabySamaniego) [inline scripts/BTLH_data_exploration.R:21]: I'm not getting errors here. It is weird.

File: scripts/BTLH_data_exploration.R [hunk 3]
```diff
@@ -14,9 +14,17 @@ raw_data <- read.csv("data/updated_raw_data.csv")
...
 
 # Remove all leading and trailing white spaces
 band_data <- mutate_all(band_data,str_trim, side=c("both"))
+# EZ: Getting an error related to non UTF-8 strings in Comment column
+# If you're also getting error, suggest replacing above with the following:
+# band_data <- band_data %>%
+#   mutate(across(!Comment, str_trim, side = "both"))
```
Comment (GabySamaniego) [inline scripts/BTLH_data_exploration.R:28]: Same here, no error.

File: scripts/BTLH_data_exploration.R (hunk 2 above)
Comment (ezylstra) [reply scripts/BTLH_data_exploration.R:21]: Interesting!  Maybe we can try and figure this (and the other similar issue) out tomorrow.

File: scripts/BTLH_data_exploration.R [hunk 4]
```diff
@@ -226,7 +237,91 @@ map1 +
...
+  labs(title = "BTLH Distribution Map with HMN sites") +
+  geom_text(data = BTLH_sites_final[4:6,],
+            aes(x = Longitude, 
+                y = nudge_lat,
+                group = NULL,
+                fill = NULL,
+                label = Location),
+            color = "black",
+            size = 3)
+
```
Comment (GabySamaniego) [inline scripts/BTLH_data_exploration.R:325]: Wow! You are right; so many details in the code for the map! Thanks!

**Review by GabySamaniego — COMMENTED** [review]
Thanks for the comments, Erin! I will update the map. This comment and those in the pull request are for practice. I hope I am not making a mess.

**Pull request merged**